### PR TITLE
[Harness] Allow to create a process manager with default tool paths

### PR DESCRIPTION
**Pull request by premun**
- `XcodeRoot` will be detected using `xcode-select` and `mlaunch` will have a hardcoded default

### CODE REVIEW

File: tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs [hunk 1]
```diff
@@ -345,5 +365,31 @@ static List<int> GetChildrenPS (ILog log, int pid)
...
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
+
+			// Something like /Applications/Xcode114.app/Contents/Developers
+			var xcodeRoot = stdout.ToString ().Trim ();
+
+			if (string.IsNullOrEmpty (xcodeRoot))
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
+
+			// We need /Applications/Xcode114.app only
+			return Path.GetDirectoryName(Path.GetDirectoryName(xcodeRoot));
```
Comment (spouliot) [inline tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs:392]: minor: missing spaces before the `(`

Comment (premun) [reply tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs:392]: True but this code will be moved and reformatted next week so I would let it go in this time

File: tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs [hunk 2]
```diff
@@ -345,5 +365,31 @@ static List<int> GetChildrenPS (ILog log, int pid)
...
+
+			var log = new MemoryLog ();
+			var stdout = new MemoryLog () { Timestamp = false };
+			var stderr = new ConsoleLog ();
+			var timeout = TimeSpan.FromSeconds (30);
+
+			var result = RunAsyncInternal (process, log, stdout, stderr, timeout).GetAwaiter ().GetResult ();
+
+			if (!result.Succeeded)
+				throw new Exception ("Failed to detect Xcode path from xcode-select!");
```
Comment (stephen-hawley) [inline tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs:383]: You have two different failure conditions that throw identical exceptions. Future you will be happier if you can distinguish the two cases.

Comment (premun) [reply tests/xharness/Microsoft.DotNet.XHarness.iOS.Shared/Execution/ProcessManager.cs:383]: Good point. Too bad it got merged already. However if this fails, we will have details in stderr (which is a ConsoleLog) so I guess we will be able to debug

**Pull request merged**